packages/react/src/components/ExpandableSearch/ExpandableSearch.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright IBM Corp. 2021
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useState, useRef, useEffect } from 'react';
+import classnames from 'classnames';
+import { settings } from 'carbon-components';
+
+import Search from '../Search';
+
+const { prefix } = settings;
+
+function ExpandableSearch(props) {
+  const [expanded, setExpanded] = useState(false);
+  const searchRef = useRef(null);
+
+  function handleFocus() {
+    if (!expanded) {
+      setExpanded(true);
+    }
+  }
+
+  function handleBlur(evt) {
+    const relatedTargetIsAllowed =
+      evt.relatedTarget &&
+      evt.relatedTarget.classList.contains(`${prefix}--search-close`);
+
+    if (
+      expanded &&
+      !relatedTargetIsAllowed &&
+      !searchRef.current.state.hasContent
+    ) {
+      setExpanded(false);
+    }
+  }
+
+  function focusInput() {
+    if (!expanded && searchRef.current?.input) {
+      searchRef.current.input.focus();
+    }
+  }
+
+  useEffect(() => {
+    if (searchRef.current?.magnifier) {
+      const { magnifier } = searchRef.current;
+      magnifier.addEventListener('click', focusInput);
+
+      return () => {
+        magnifier.removeEventListener('click', focusInput);
+      };
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Feel free to inline that `focusInput` helper to the `useEffect` since it seems like we're only using it there 👀  Should help with the eslint violation without much cost.

Reply:
Thanks for the tip!